moveit_ros/moveit_servo/demos/cpp_interface/demo_pose.cpp

```diff
@@ -0,0 +1,143 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+/*      Title     : demo_pose.cpp
+ *      Project   : moveit_servo
+ *      Created   : 06/07/2023
+ *      Author    : V Mohammed Ibrahim
+ *      Description : Example of controlling a robot through pose commands via the C++ API.
+ */
+
+#include <chrono>
+#include <atomic>
+#include <mutex>
+#include <rclcpp/rclcpp.hpp>
+#include <moveit_servo/servo.hpp>
+#include <moveit_servo/utils/common.hpp>
+
+using namespace moveit_servo;
+
+namespace
+{
+const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_servo.pose_demo");
+}
+
+int main(int argc, char* argv[])
+{
+  rclcpp::init(argc, argv);
+
+  // The servo object expects to get a ROS node.
+  auto demo_node = std::make_shared<rclcpp::Node>("moveit_servo_demo");
+
+  // Get the servo parameters.
+  std::string param_namespace = "moveit_servo";
+  auto servo_param_listener = std::make_shared<const servo::ParamListener>(demo_node, param_namespace);
+  auto servo_params = servo_param_listener->get_params();
+
+  // The publisher to send trajectory message to the robot controller.
+  rclcpp::Publisher<trajectory_msgs::msg::JointTrajectory>::SharedPtr trajectory_outgoing_cmd_pub =
+      demo_node->create_publisher<trajectory_msgs::msg::JointTrajectory>(servo_params.command_out_topic,
+                                                                         rclcpp::SystemDefaultsQoS());
+
+  // Create the servo object
+  auto planning_scene_monitor = createPlanningSceneMonitor(demo_node, servo_params);
+  auto servo = Servo(demo_node, servo_param_listener, planning_scene_monitor);
+
+  // Wait for some time, so that the planning scene is loaded in rviz.
+  // This is just for convenience, should not be used for sync in real application.
+  std::this_thread::sleep_for(std::chrono::seconds(3));
+
+  // For syncing pose tracking thread and main thread.
+  std::mutex pose_guard;
+  std::atomic<bool> stop_tracking = false;
+
+  // Set the command type for servo.
+  servo.expectedCommandType(CommandType::POSE);
+
+  // The dynamically updated target pose.
+  PoseCommand target_pose;
+  target_pose.frame_id = servo_params.ee_frame;
+  target_pose.pose = Eigen::Isometry3d::Identity();  // Target pose in ee frame (same as the ee frame pose)
+  // Convert the pose to planning frame.
+  target_pose = servo.toPlanningFrame(target_pose);
+
+  // The pose tracking lambda that will be run in a separate thread.
+  auto pose_tracker = [&]() {
+    KinematicState joint_state;
+    rclcpp::WallRate tracking_rate(1 / servo_params.publish_period);
+    while (!stop_tracking)
```

Comment:
To ensure Ctrl+C still functions. Please check if this change is needed anywhere else.

```suggestion
    while (!stop_tracking && rclcpp::ok())
```

Reply:
Done